subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -214,13 +215,19 @@ public void execute(ExecutionListener executionListener) {
                 ParsedCommandLine parsedCommandLine = parser.parse(args);
 
                 buildLayoutConverter.convert(parsedCommandLine, buildLayout);
-                loggingConfigurationConverter.convert(parsedCommandLine, loggingConfiguration);
+
 
                 Map<String, String> properties = new HashMap<String, String>();
                 // Read *.properties files
                 layoutToPropertiesConverter.convert(buildLayout, properties);
                 // Read -D command line flags
                 systemPropertiesCommandLineConverter.convert(parsedCommandLine, properties);
+
+                // Convert properties for logging  object
+                PropertiesToLogLevelConfigurationConverter propertiesToLogLevelConfigurationConverter = new PropertiesToLogLevelConfigurationConverter();
+                propertiesToLogLevelConfigurationConverter.convert(properties, loggingConfiguration);
+                loggingConfigurationConverter.convert(parsedCommandLine, loggingConfiguration);
+
```

Comment:
:o: You could look at [this test](https://github.com/gradle/gradle/blob/master/subprojects/launcher/src/integTest/groovy/org/gradle/launcher/CommandLineIntegrationSpec.groovy#L215-L241).